File: code_snippets/api-guide/model/deep_training/training_time_estimator.java
package com.clarifai.example;

import com.clarifai.grpc.api.*;
import com.clarifai.channel.ClarifaiChannel;
import com.clarifai.credentials.ClarifaiCallCredentials;
import com.clarifai.grpc.api.status.StatusCode;
import com.google.protobuf.Struct;
import com.google.protobuf.Value;

public class ClarifaiExample {

    //////////////////////////////////////////////////////////////////////////////////////////////////////
    // In this section, we set the user authentication, app ID, model ID, and estimated input count.
    // Change these strings to run your own example.
    /////////////////////////////////////////////////////////////////////////////////////////////////////

    static final String USER_ID = "YOUR_USER_ID_HERE";
    // Your PAT (Personal Access Token) can be found in the portal under Authentication
    static final String PAT = "YOUR_PAT_HERE";
    static final String APP_ID = "YOUR_APP_ID_HERE";
    // Change these to get your training time estimate
    static final String MODEL_ID = "YOUR_CUSTOM_MODEL_ID_HERE";
    static final int ESTIMATED_INPUT_COUNT = 100;

    ///////////////////////////////////////////////////////////////////////////////////
    // YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
    ///////////////////////////////////////////////////////////////////////////////////	

    public static void main(String[] args) {
        
        V2Grpc.V2BlockingStub stub = V2Grpc.newBlockingStub(ClarifaiChannel.INSTANCE.getGrpcChannel())
                .withCallCredentials(new ClarifaiCallCredentials(PAT));
        
        Struct.Builder params = Struct.newBuilder()
                .putFields("template", Value.newBuilder().setStringValue("MMDetection_FasterRCNN").build());
        
        MultiTrainingTimeEstimateResponse trainingTimeEstimateResponse = stub.postModelVersionsTrainingTimeEstimate(
                PostModelVersionsTrainingTimeEstimateRequest.newBuilder()
                        .setUserAppId(UserAppIDSet.newBuilder().setUserId(USER_ID).setAppId(APP_ID))
                        .setModelId(MODEL_ID)
                        .addModelVersions(ModelVersion.newBuilder()
                                .setTrainInfo(TrainInfo.newBuilder()
                                        .setParams(params)
                                )
                        )
                        .setEstimatedInputCount(ESTIMATED_INPUT_COUNT)
                        .build()
        );
        
        if (trainingTimeEstimateResponse.getStatus().getCode() != StatusCode.SUCCESS) {
            throw new RuntimeException("Post model outputs failed, status: " + trainingTimeEstimateResponse.getStatus());
        }
        
        System.out.print(trainingTimeEstimateResponse);
        
    }
}

File: code_snippets/api-guide/model/deep_training/training_time_estimator.js
//index.js file

//////////////////////////////////////////////////////////////////////////////////////////////////////
// In this section, we set the user authentication, app ID, model ID, and estimated input count.
// Change these strings to run your own example.
/////////////////////////////////////////////////////////////////////////////////////////////////////

const USER_ID = "YOUR_USER_ID_HERE";
// Your PAT (Personal Access Token) can be found in the Account's Security section
const PAT = "YOUR_PAT_HERE";
const APP_ID = "YOUR_APP_ID_HERE";
// Change these to get your training time estimate
const MODEL_ID = "YOUR_CUSTOM_MODEL_ID_HERE";
const ESTIMATED_INPUT_COUNT = 100;

///////////////////////////////////////////////////////////////////////////////////
// YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
///////////////////////////////////////////////////////////////////////////////////

const { ClarifaiStub, grpc } = require("clarifai-nodejs-grpc");

const stub = ClarifaiStub.grpc();

// This will be used by every Clarifai endpoint call
const metadata = new grpc.Metadata();
metadata.set("authorization", "Key " + PAT);

stub.PostModelVersionsTrainingTimeEstimate(
  {
    user_app_id: {
      user_id: USER_ID,
      app_id: APP_ID,
    },
    model_id: MODEL_ID,
    model_versions: [{
      train_info: {
        params: {
          template: "MMDetection_FasterRCNN"
        }
      },

    }],
    estimated_input_count: ESTIMATED_INPUT_COUNT

  },

  metadata,

  (err, response) => {
    if (err) {
      throw new Error(err);
    }

    if (response.status.code !== 10000) {
      console.error("Request failed, status: ", response.status);
      throw new Error("Request failed, status: " + response.status.description);
    }

    console.log(response);

  }
);